modules/cardano-service.nix

```diff
@@ -43,7 +43,7 @@ let
     "--logs-prefix /var/lib/cardano-node"
     "--db-path ${stateDir}/node-db"
     (optionalString (!cfg.enableP2P) "--kademlia-explicit-initial --disable-propagation ${smartGenPeer}")
-    "--configuration-file ${cardano.src}/configuration.yaml"
+    "--configuration-file ${cardano.src + "/../lib/"}/configuration.yaml"
```

Comment:
Did you test this? I'm sure this doesn't work at all. You'll have to do `cardano-sl.src` here.

Reply:
The change is not mine, it was suggested by @pva701, but I did test it, and it worked perfectly fine for me.

Reply:
What did you test? Because I'm almost sure this can't work once service has started.

Reply:
I successfully managed to deploy a cluster with a pre-generated genesis by following "Developer clusters HOWTO" instruction.

Reply:
Interesting, can you paste what arguments does cardano-sl have on those machines?

Reply:
I'm not exactly sure how to get them, but here are lines from one of nodes' log:
```
Jan 12 16:20:20 csl-1983.ec2.a1 cardano-node-start[2760]: [*production*:INFO:ThreadId 4] [2018-01-12 16:20:20.31 UTC] using configurations: ConfigurationOptions {cfoFilePath = "/nix/store/pm7lkdhi6s2bc51ykg3l963aii3rqgqy-lib/configuration.yaml", cfoKey = "internal_staging_full", cfoSystemStart = Just 1515773787000000, cfoSeed = Nothing}
```

Reply:
Okay, here's the full command of one of the nodes
```
/nix/store/y5q6700jxpv134r8pwkp0a9f1aqdnxd7-cardano-sl-node-1.0.3/bin/cardano-node-simple --address 35.156.158.135:3000 --listen 172.31.31.105:3000 --no-ntp --json-log /var/lib/cardano-node/jsonLog.json --metrics +RTS -T -RTS --statsd-server 127.0.0.1:8125 --ekg-server 172.31.31.105:8080 --keyfile /var/lib/cardano-node/key0.sk --system-start 1515774707 --log-config /nix/store/wh6z2ksmb042vc5r8ar9m75ry7b2lcfy-csl-logging.yaml --logs-prefix /var/lib/cardano-node --db-path /var/lib/cardano-node/node-db --configuration-file /nix/store/pm7lkdhi6s2bc51ykg3l963aii3rqgqy-lib/configuration.yaml --configuration-key internal_staging_full --topology /nix/store/k5aqs3c5lfbc5fsw5vazr408fsggmp93-topology-development.yaml --node-id a1
```

Reply:
```
nix-repl> cardano-sl-node.src + "/../lib/"
/home/domenkozar/dev/cardano-sl/lib
```

Ah! This gets me every time.